src/Nethermind/Nethermind.Network/P2P/Subprotocols/Eth/V65/Eth65ProtocolHandler.cs

```diff
@@ -66,11 +68,43 @@ GetPooledTransactionsMessage getPooledTxMsg
                     Handle(getPooledTxMsg);
                     break;
                 case Eth65MessageCode.NewPooledTransactionHashes:
-                    Metrics.Eth65NewPooledTransactionHashesReceived++;
+                    NewPooledTransactionHashesMessage newPooledTxMsg =
+                        Deserialize<NewPooledTransactionHashesMessage>(message.Content);
+                    ReportIn(newPooledTxMsg);
+                    Handle(newPooledTxMsg);
                     break;
             }
         }
 
+        private void Handle(NewPooledTransactionHashesMessage msg)
+        {
+            Metrics.Eth65NewPooledTransactionHashesReceived++;
+            Stopwatch stopwatch = Stopwatch.StartNew();
+            List<Keccak> discoveredTxHashes = new();
+            
+            for (int i = 0; i < msg.Hashes.Count; i++)
+            {
+                Keccak hash = msg.Hashes[i];
+                if (!_txPool.IsInHashCache(hash))
+                {
+                    if (_txPool.TryAddToPendingHashes(hash))
+                        discoveredTxHashes.Add(hash);
+                }
+            }
+
+            int discoveredTxHashesCount = discoveredTxHashes.Count;
+
+            if (discoveredTxHashesCount != 0)
+            {
+                Send(new GetPooledTransactionsMessage(discoveredTxHashes.ToArray()));
```

Comment:
extract to a separate class that has only one responsibility -> requesting mising transactions, cover that class with unit tests, hide behind interface

Reply:
we have added a responsibility of transaction fetching to the handler which is undesired